axelor-sale/src/main/java/com/axelor/apps/sale/rest/SaleOrderRestController.java

```diff
@@ -63,4 +70,25 @@ public Response createSaleOrder(SaleOrderPostRequest requestBody)
 
     return ResponseConstructor.buildCreateResponse(saleOrder, new SaleOrderResponse(saleOrder));
   }
+
+  @Operation(
+      summary = "Update sale order status",
+      tags = {"Sale order"})
+  @Path("/status")
+  @PUT
+  @HttpExceptionHandler
+  public Response changeSaleOrderStatus(SaleOrderPutRequest requestBody) throws AxelorException {
+    RequestValidator.validateBody(requestBody);
+    new SecurityCheck().writeAccess(SaleOrder.class, requestBody.getSaleOrderId()).check();
+    String toStatus = requestBody.getToStatus();
+    SaleOrder saleOrder = requestBody.fetchSaleOrder();
+    if (SaleOrderPutRequest.SALE_ORDER_UPDATE_FINALIZE.equals(toStatus)) {
+      Beans.get(SaleOrderFinalizeService.class).finalizeQuotation(requestBody.fetchSaleOrder());
+    }
+    if (SaleOrderPutRequest.SALE_ORDER_UPDATE_CONFIRM.equals(toStatus)) {
+      Beans.get(SaleOrderConfirmService.class)
+          .confirmSaleOrder(Beans.get(SaleOrderRepository.class).find(saleOrder.getId()));
+    }
```

Comment:
You are using not using the saleOrder variable you created when using the 2 services methods.

Also you should use ObjectFinder when working with API PUT request to have a version control :
Example : Timesheet timesheet = ObjectFinder.find(Timesheet.class, timesheetId, requestBody.getVersion());